sdk/core/abort-controller/src/AbortSignal.ts

```diff
@@ -143,12 +176,15 @@ export class AbortSignal implements AbortSignalLike {
  *
  * @internal
  */
-// eslint-disable-next-line @azure/azure-sdk/ts-use-interface-parameters
-export function abortSignal(signal: AbortSignal): void {
+// eslint-disable-next-line @azure/azure-sdk/ts-use-interface-parameters,@typescript-eslint/explicit-module-boundary-types
+export function abortSignal(signal: AbortSignal, reason?: any): void {
   if (signal.aborted) {
     return;
   }
 
+  // if not specified, abortSingal.reason defaults to AbortError
+  signal["_reason"] = reason ?? new AbortError("The operation was aborted.");
```

Comment:
feels bad to reach into private state from a helper... could we use another `WeakMap` like we do for listeners and the aborted state instead?

Reply:
Yes using WeakMap is cleaner and more consistent